sktime/utils/_maint/_show_versions.py

```diff
@@ -52,28 +52,53 @@ def _get_sys_info():
     "tslearn",
     "torch",
     "tensorflow",
-    "tensorflow_probability",
 ]
 
 
-def _get_deps_info(deps=None):
+def _get_deps_info(deps=None, source="distributions"):
     """Overview of the installed version of main dependencies.
 
     Parameters
     ----------
-    deps : optional, list of strings with import names
-        if None, behaves as deps = ["sktime"]
+    deps : optional, list of strings with package names
+        if None, behaves as deps = ["sktime"].
+
+    source : str, optional one of "importlib" (default) or "version"
+        source of version information
+
+        * "distributions" - uses importlib.distributions. In this case,
+          strings in deps are assumed to be PEP 440 package strings,
+          e.g., scikit-learn, not sklearn.
+        * "import" - uses the __version__ attribute of the module.
+          In this case, strings in deps are assumed to be import names,
+          e.g., sklearn, not scikit-learn.
 
     Returns
     -------
     deps_info: dict
         version information on libraries in `deps`
-        keys are import names, values are PEP 440 version strings
+        keys are package names, import names if source is "version",
+        and PEP 440 package strings if source is "importlib";
+        values are PEP 440 version strings
         of the import as present in the current python environment
     """
     if deps is None:
         deps = ["sktime"]
 
+    if source == "distributions":
```

Comment:
If this option is safe and work always, is there any reason to keep the choice? It's an internal private function.

(Non-blocking comment for discussion)

Reply:
My reasoning was to be able test one against the other, since currently that is what happens when we test `show_versions` for consistency with `_check_soft_dependencies`. If we refactor, this sanity check would be removed.

However, I have not actually changed the test to retain coverage of this sanity check (which I notice, I probably got side tracked with release, admin, and non-sktime professional duties). This should be added in the tests.